### PR TITLE
Add simple file sink

### PR DESCRIPTION
**Pull request by gurug-prog**

### CODE REVIEW

File: lib/sinks/file-sink.js [hunk 1]
```diff
@@ -1,7 +1,35 @@
...
+'use strict';
+
+const fs = require('fs').promises;
+const StringBuilder = require('../string-builder');
+
+
+// add chaining syntax returning (this) in StringBuilder
+// here we maybe can iterate asynchronously
+const buildMessage = (loggingLevel, messageParts, entities) => {
+  const spaceCount = 2;
```
Comment (MatthewKirik) [inline lib/sinks/file-sink.js:10]: make this constant module-wise

File: lib/sinks/file-sink.js [hunk 2]
```diff
@@ -1,7 +1,35 @@
...
+  const sb = new StringBuilder();
+  for (let i = 0; i < messageParts.length - 1; i++) {
+    sb.append(messageParts[i]);
+    if (i === messageParts.length - 1) {
+      sb.append(`\n#${loggingLevel}`);
```
Comment (MatthewKirik) [inline lib/sinks/file-sink.js:15]: why using # symbol? It means nothing in the terms of filesystem and is only useful for telegram sink.
You can change format to "LOGGING_LEVEL [DateTime]: Logging message"

File: lib/sinks/file-sink.js [hunk 3]
```diff
@@ -1,7 +1,35 @@
...
+    } else {
+      sb.append('\n' + JSON.stringify(
+        entities[i], null, spaceCount) + '\n'); // why not?
+    }
+    // message += messageParts[i] + "\n";
```
Comment (MatthewKirik) [inline lib/sinks/file-sink.js:20]: Don't push unused code.

File: lib/sinks/file-sink.js [hunk 4]
```diff
@@ -1,7 +1,35 @@
...
+  }
+
+  return sb.toString();
+};
 
 const createSink = (options) => {
-  const sink = (loggingLevel, messageParts, entities) => {};
+  const sink = (loggingLevel, messageParts, entities) => {
+    buildMessage(loggingLevel, messageParts, entities);
+
```
Comment (MatthewKirik) [inline lib/sinks/file-sink.js:31]: Put file-writing code here

File: lib/string-builder.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    // const str = this.strings[i];
+    return this.strings.join('');
+  }
+}
+
+module.exports = StringBuilder;
+
+
+
+// class StringBuilder {
```
Comment (MatthewKirik) [inline lib/string-builder.js:22]: Don't push unused code

File: lib/string-builder.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+class StringBuilder {
+  constructor() {
+    this.strings = [];
+  }
+
+  append(str) {
```
Comment (MatthewKirik) [inline lib/string-builder.js:8]: Return 'this' for OOP-style chaining.

**Review by MatthewKirik — COMMENTED** [review]
Merge is useless as the changes do not contain the core code of the feature that it intents to implement. Please, write the needed code first and make corrections to match with the requirements.

**Comment by gurug-prog** [comment]
Sorry, I've pushed wrong branch from my old PC. I will push up-to-date branch soon. Now I'm closing my PR.

**Pull request closed without merging**